%	This is written by Zhiyang Ong as a template for writing reports.

%	The MIT License (MIT)

%	Copyright (c) <2014> <Zhiyang Ong>

%	Permission is hereby granted, free of charge, to any person obtaining a copy of this software and associated documentation files (the "Software"), to deal in the Software without restriction, including without limitation the rights to use, copy, modify, merge, publish, distribute, sublicense, and/or sell copies of the Software, and to permit persons to whom the Software is furnished to do so, subject to the following conditions:

%	The above copyright notice and this permission notice shall be included in all copies or substantial portions of the Software.

%	THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.

%	Email address: echo "cukj -wb- 23wU4X5M589 TROJANS cqkH wiuz2y 0f Mw Stanford" | awk '{ sub("23wU4X5M589","F.d_c_b. ") sub("Stanford","d0mA1n"); print $5, $2, $8; for (i=1; i<=1; i++) print "6\b"; print $9, $7, $6 }' | sed y/kqcbuHwM62z/gnotrzadqmC/ | tr 'q' ' ' | tr -d [:cntrl:] | tr -d 'ir' | tr y "\n"

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%



%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%	Preamble.
\documentclass[letter,12pt]{article}
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%
%	Importing LaTeX source files, without quoting the ".tex" extension.
%
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%	File containing the LaTeX preamble.
\input{./others/preamble_section}





% definition of new \LaTeX command for the citation: \cite{Cimatti08} and \cite{Barrett09}
% This allows mathematical/logic symbols to be typeset with the font ``Zapf Chancery'' in ``\LaTeX\ math mode''. To typeset symbols in such font, try: \mathpzc{ABCdef123}
\DeclareMathAlphabet{\mathpzc}{OT1}{pzc}{m}{it}

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
% Start of document
\begin{document}
\title{Homework for June 26, 2014}
\date{\today}
\author{Zhiyang Ong\thanks{Email correspondence to: \href{mailto:ongz@acm.org}{\Email\ ongz@acm.org}}\\
	Texas A\&M University
}
\maketitle


\begin{abstract} 
This is the homework for June 26, 2014. It contains a description of the comparison of different SAM files. It also contains hypotheses for why the given SRA82660 sequences do not map to chromosome 7 (linkage group VII) of the {\it Neurospora crassa} species.
\end{abstract}




%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
\section{Acknowledgements*}
\label{sec:Acknowledgement*}

I acknowledge that I have collaborated with my project teammates Diana Medina and Selene Howe to complete this homework. Specifically, we discussed some hypotheses. \\

I also acknowledge the help that I have received from the course instructor, Prof. Rodolfo Aramayo, in getting back to my queries about the source of the reference sequence for the {\it Neurospora crassa} species that was provided to us on Tuesday. \\

Lastly, I want to thank Ms. Tess L. Pham from the Tutor Zone at Evans Library, which is run by the Academic Success Center at Texas A\&M University. Ms. Pham has tutored me about the basics of genetics and gave me feedback about the hypotheses which I proposed. I also want to thank Ms. Priyadharshini Venkat, my classmate for this course, for giving me feedback about my hypotheses and helping me to find alternate golden models for the {\it Neurospora crassa} species.

  
%	Student Learning Center

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
\section{Comparison of different SAM files}
\label{sec:ComparisonofDifferentSAMfiles}

Different types of SAM files have to be compared. Their differences are only based on the different combinations of {\tt -v} and {\tt -m} selected, or different combinations of {\tt -n} and {\tt -m} selected for {\it -Bowtie} \cite{Langmead2014}. \\

The different combinations of these {\tt -v}, {\tt -n}, and {\tt -m} options of {\it -Bowtie} are: \vspace{-0.3cm}
\begin{enumerate} \itemsep -4pt
\item -v1 -m1
\item -v1 -m100
\item -v3 -m1
\item -v3 -m100
\item -n1 -m1
\item -n1 -m100
\item -n3 -m1
\item -n3 -m100
\end{enumerate}

The {\tt -v} option is used to determine the number of alignments that have no more than ``v'' mismatches. This addresses global alignments, where a sequence is matched with another. The {\tt -m} option reports alignments with a maximum of ``m'' repeats. The {\tt -n} option reports alignments with a maximum of ``m'' repeats. This addresses local alignments, where a region in a sequence is matched with another region in the sequence \cite{Aramayo2014}.

\begin{table}[htdp]
\caption{Bowtie results for SRA82660}	\vspace{-0.2in}
\label{tab:bowtie}
	\begin{center}
		\begin{tabular}{|c|c|c|c|c|c|c|c|c|}
		\hline
		\# reads (out of 86036) with at & 5987 & 5995 & Abstraction \\
		least n reported alignments & & & & &
		\hline
		\# reads that & 80041 & Features & Abstraction \\
		failed to align & 93.0\%
		\hline
		\# reads with alignments  & Use & Features & Abstraction \\
		suppressed due to -m & 
		\hline
		Reported \# paired-end alignments & 
		\hline
		\end{tabular}
	\end{center}
\end{table}




%	JHU
%	Baltimore, Maryland





%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
\section{Hypothesis: SRA82660 sequence is not part of the {\it Neurospora crassa} genome}
\label{sec:Hypothesis}


A hypothesis is that the SRA82660 sequence is not part of the {\it Neurospora crassa} genome. That is, it belongs to another species. Proceedure \ref{lst:SRA82660notNeurosporacrassa}
%genome sequence is wrong.


\begin{codebox}
\Procname{$\proc{Test if SRA82660 sequence is part of the {\it Neurospora crassa} genome}({\it Golden/Reference Model})$}
\label{lst:SRA82660notNeurosporacrassa}
\zi \Comment {\it Golden/Reference Model}
%\zi \Comment {\it Input ARGUMENT \#2: Definition}
\li	\If $[$given SRA82660 subsequence does not map to golden model$]$
	\Then
\li		Conclusion: The given SRA82660 subsequence is not part of the {\it Neurospora crassa} species.
%	\li		Conclusion \#1: The given SRA82660 subsequence may not have been processed
%	\zi		\> correctly from the SRA82660 sequence in the NCBI database.
%	\li		Conclusion \#2: The given SRA82660 sequence in the NCBI database has errors.
\li	\ElseIf $[$given SRA82660 subsequence maps to golden model$]$
	\Then
\li		Conclusion: The given SRA82660 subsequence is part of the genome.
\li	
%\li		Conclusion \#2: The given SRA82660 subsequence does not contain chromosome 7
%\li	\ElseNoIf $[$Condition$]$
%\li	
%\li	\Else
%\li	
	\End
%\li BODY OF THE PROCEDURE
%\zi	\Comment {\it What is the output of this procedure?}
\li	\Return\end{codebox}


















%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%
%	End of document
%
%	Inserting references
%
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%	Beginning of BACK MATTER: bibliography, indexes and colophon
%\backmatter

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
{\linespread{1}
\bibliographystyle{plain}
%\bibliography{./references/references}
\bibliography{/data/research/antipastobibtex/references}
}
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
\end{document}